File: 프로그래머스/lv2/64065. 튜플/튜플.js
function solution(s) {
    const answer = [];
    
    s = s.slice(1, s.length - 1);
    
    const result = [];
    let tmp = [];
    let tmpString = '';
    for (const item of s) {
        if (item === '}') {
            if (tmpString) {
                tmp.push(+tmpString);
                tmpString = '';
            }
            result.push(tmp);
            tmp = [];
        }
        else if (item === ',' && tmpString) {
            tmp.push(+tmpString);
            tmpString = '';
        }
        
       else if (item !== '{' && item !== ',') {
           tmpString += item;
       } 
    }
    
    result.sort((a, b) => {
        return a.length - b.length
    })
    
    for (const item of result) {
        for (let i = 0; i < item.length; i++) {
            if (!answer.length) answer.push(item[i]);
            else {
                if (!answer.includes(item[i])) answer.push(item[i]);
            }
        }
    }
    
    return answer;
}

File: 프로그래머스/lv2/64065. 튜플/튜플.py
def solution(s):
    s = s[1:len(s)-1].replace("{", "").split("},")
    s[len(s) - 1] = s[len(s) - 1].replace("}", "")
    s_array = []
    for items in s:
        items = list(map(int, items.split(",")))
        s_array.append(items)
    
    answer = []
    s_array = sorted(s_array, key = lambda x : len(x))
    
    for items in s_array:
        for item in items:
            if item not in answer:
                answer.append(item)

    return answer